service/remotecfg/remotecfg.go

```diff
@@ -133,6 +229,36 @@ func (s *Service) Run(ctx context.Context, host service.Host) error {
 // Update implements [service.Service] and applies settings.
 func (s *Service) Update(newConfig any) error {
 	newArgs := newConfig.(Arguments)
+
+	if newArgs.URL == "" {
+		// TODO: We either never set the block on the first place,
+		// or recently removed it. Make sure we stop everything gracefully
+		// before returning.
+		return nil
+	}
+
+	// TODO: Should we also include the username, or some other property on the
+	// hash for the directory name?
+	s.dataPath = filepath.Join(s.opts.StoragePath, ServiceName, getHash(newArgs.URL))
+	s.ticker.Reset(newArgs.PollFrequency)
```

Comment:
The way this works means that you won't get the new config until after the `PollFrequency` elapses, which might be confusing to users who reloaded their config (imagine changing the frequency to `1h` but it takes one hour to get the first config after the reload).

I think calling Update should immediately trigger a new retrieval.

Reply:
I've refactored the fetch-from-api-and-fallback-to-cache logic in a new method that's called from Update.